synapseutils/sync.py

```diff
@@ -116,9 +116,10 @@ def generateManifest(syn, allFiles, filename):
     annotKeys = set()
     data = []
     for entity in allFiles:
-        row = {'parent': entity['parentId'], 'path': entity.path, 'name': entity.name,
+        row = {'parent': entity['parentId'], 'path': entity.get("path"), 'name': entity.name,
                'synapseStore': entity.synapseStore, 'contentType': allFiles[0]['contentType']}
-        row.update({key:val[0] for key, val in entity.annotations.items()})
+        row.update({key:val[0] for key, val in entity.annotations.items() if len(val) > 0})
+        row.update({key:"" for key, val in entity.annotations.items() if len(val) == 0})
```

Comment:
you can avoid iterating the annotations twice with
`row.update({key:(val[0] if len(val) > 0 else "") for key, val in entity.annotations.items()})`